lib/srv/db/cloud/users/helpers.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package users
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/secrets"
+	"github.com/gravitational/teleport/lib/srv/db/common"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+// lookupMap is a mapping of database objects to their managed users.
+type lookupMap struct {
+	byDatabase map[types.Database][]User
```

Comment:
Do we need to track users per database if a mange users per database can be obtained 
 by the `types.Database.GetManagedUsers()`call ?

Reply:
currently `GetManagedUsers` only return a list of user names. We need to call get password for that user when `tsh db connect`. Database username is not unique across databases so I needed this map to find the `User` object.

Though `GetManagedUsers` can work if it returns a proper interface that has like GetName and GetPassword etc. functions. I did hesitate to add that in api/types.

What do you think?

Reply:
Ah, the I think that is is reasonable to keep this mapping in code.